thorlcr/thorutil/thmem.cpp

```diff
@@ -341,17 +341,14 @@ class CSharedSpillableRowSet : public CSpillableStreamBase, implements IInterfac
     }
     IRowStream *createRowStream()
     {
+        CRowsLockBlock block(*this);
```

Comment:
@richardkchapman - this is the significant change/fix.
It is moving the row lock into method scope, out from the block that it used to be in.
Thereby it ensures that it is blocked or locked at the time CStream is constructed (and registerWriteCallback is called)